### DISCUSSION TITLE
Koishi 4.17.0-alpha.0

### DISCUSSION THREAD

**Discussion by shigma** [koishijs/koishi#1361] [General]
这是一个预发布版本，主要涉及底层架构调整，并未包含新功能。

## Notable Changes

- **cordis:** 引入了与 Koishi 类似的 Loader / HMR 等功能，现已实装于 Yakumo 等框架中，并将在未来替换 Koishi 的底层实现 (cordiverse/cordis@413a99911429a9508fc59bbcef0e9bf8727366c4, cordiverse/cordis@fd408a422375972b4ff02a9647063919d055b133, cordiverse/cordis@f58cb365be90006e99a1362ec8e66bd8bce9b3c4)
- **cordis:** 将核心功能发布到 @cordisjs/core，而 cordis 将自带 Timer 和 Logger 服务 (cordiverse/cordis@94b114a33e1581c499379dca8c5a026d755ffde3, cordiverse/cordis@1efd00be910ce5a46d6642e8a451efcf78d1aeb6, cordiverse/cordis@061043b1cb1f0bf9f29f13f270bf02700a677c9d, cordiverse/cordis@2a7a9479eded1fb9419508a3999ed9dada2c9b49, satorijs/satori@3ff856a5fc559b61a0839c014b85dfaeb3e4b9b6, satorijs/satori@8f9662560b27e55a8e9e57f927377d38fd8ea5f4)
- **minato:** 现在 Minato 也将作为 Cordis 生态的一部分，可以直接作为 Cordis 插件使用 (cordiverse/minato@aef9dbad9cf3c9abc56cdd3b2104f2757d19793d, cordiverse/minato@d889604dd052b5a6c0f5335b0753f5a2aadd0ddb, cordiverse/minato@228fff5fc14a3772bb2d9d4eaaded4a09fb34c53, cordiverse/minato@cdf830950a5fb651c8e26b3617dfdb0c00e8e5ef)
- **minato:** 为 Minato 侧的数据库插件添加了配置约束，同时移除了一些与 Koishi 耦合的默认值；这些默认值将在新版本 Koishi 启动时自动写入配置文件，以保证向下兼容性 (cordiverse/minato@25ed14eba9020f6acc3f5f082b4ac8f455e04c49, cordiverse/minato@99b2baf7574c8ea8effb7e33625a42830956b51b, koishijs/koishi@61c7d2c8579ca48bf18cdcf935d82cace8374f02)

## Features

- **cordis:** 简化了 Cordis 的类型，修复了过度使用泛型导致派生框架类型报错的问题 (cordiverse/cordis@e34e9cf588d32f2b34e5f6bd9c58cc93517f9dc7)  (cordiverse/cordis@c1ac7b0b6f61002d6b96089ee5527c50828aa8b4)
- **cordis:** 支持了 `Service` 类不传入一参数 `ctx` 的情况，此时将自动创建一个根上下文 (cordiverse/cordis@e0d96b4372b111ea9209ce4d79c0e69af0281f0f)
- **cordis:** 新增了 `internal/info` 事件 (cordiverse/cordis@159742e3f3a25d9d93196397a8435b8464316f74)
- **minato**: 新增 `driver.dropAll()` 方法，用于对应实现 `database.dropAll()` (cordiverse/minato@fb984b3334b54503403e7a73f0b79c66d9710f84)
- **qq:** 引入了更多输出日志 (satorijs/satori#236) (satorijs/satori@b0f6d63769bc21c2127a00fb9724cec855114ed9)
- **tsc:** 生成的 `.dts` 文件将根据 `exports` 决定 (cordiverse/yakumo@f0c742a26ab77616cdc8fd2a7a8a9832f351e6bc)

## Bug Fixes

- **cordis:** 修复了使用 `ctx.effect()` 创建的副作用不会自动回收的问题 (cordiverse/cordis#8) (cordiverse/cordis@957e74d9ade9da0f6e51c3f43830a38392858ab4)
- **discord:** 修复了 `bot.getMessage()` 多次或递归触发会导致 429 的问题 (satorijs/satori@e5825f14c5847dbd3226ad1d01c8d015ce5fa509)

## Other Changes

- chore: bump versions (koishijs/koishi@6cc2f9b972b262fdfdc6f938c92bca29ba1570ed)
- chore: refactor to monorepo (cordiverse/cordis@21a86a35e12d563ebfb1b665f659c4a3bdf1ee87)
- chore: patch tsx to fix CI error (cordiverse/cordis@502e63f9e9945f05eb95abb6db7d8ec138105576)
- chore: remove redundant dependencies (cordiverse/cordis@d3eec26738e8261c358bd4b425f4552de3e40fc8)
- chore: bump versions (cordiverse/yakumo@970ea6d136ade9f2ab09c469e7900db77140065c)
- refa: temporarily allow importing ts extensions (cordiverse/cordis@bdd36a0f1b8a18498814e1db11813dff5f5cbcd5)
- types: use `Options` instead of `Config` if it is not actual config (satorijs/satori@979f2da254512680ff7142754630749bc25b6526)
- chore: bump versions (satorijs/satori@27822a0d4412a824ac46594fa2681a0e2d1109d5)
- chore: add database-* plugins (koishijs/upstream@9a69ef8520cc0517c111252d26251576ef443dbb)
- chore: bump versions (koishijs/upstream@b2850a5764279acecb7cb8ab2767ea4680def5a1)
- chore: migrate database-* plugins to upstream (koishijs/koishi@64e13d07a61ca5e3184f7399a2b910f76b8a6967)

<hr /><em>This discussion was created from the release <a href='https://github.com/koishijs/koishi/releases/tag/4.17.0-alpha.0'>Koishi 4.17.0-alpha.0</a>.</em>

**Comment by MaikoTan** [comment]
On new year's eve, Shigma was still coding away, forgotten previous content, She's really a crybaby.

除夕夜梦梦还在写代码，前面忘了，她真的，我哭死。

**Comment by LeeJSwit-HCN** [comment]
辛苦



---Original---
From: "Maiko Sinkyaet ***@***.***&gt;
Date: Fri, Feb 9, 2024 23:37 PM
To: ***@***.***&gt;;
Cc: ***@***.***&gt;;
Subject: Re: [koishijs/koishi] Koishi 4.17.0-alpha.0 (Discussion #1361)




 
On new year's eve, Shigma was still coding away, forgotten previous content, She's really a crybaby.
 
除夕夜梦梦还在写代码，前面忘了，她真的，我哭死。
